.github/workflows/codeql.yml

```diff
@@ -0,0 +1,40 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Cache Maven packages
+      uses: actions/cache@v3
```

Comment:
🙃 
```suggestion
      uses: actions/cache@v4
```

Reply:
I can give it a try, but from what I tested now, this caching is only making the build like 30 seconds faster (out of 20 minutes), so I'm probably gonna remove it anyway.

Reply:
Maybe the effect will only show after a couple of runs when more of it was cached.
If you decide to go with caching, I would say let's use the latest version.
This should not block you from making progress now. :)

Reply:
I tried doing several runs indeed and saw no enhancement :( Will do more tests.
Done, updated to v4.